### ISSUE TITLE
Deep4Net starts in eval mode

### ISSUE THREAD

**Issue by dungscout96** [braindecode/braindecode#744]
It looks like Deep4Net is put into eval mode by default: https://github.com/braindecode/braindecode/blob/32d44b7ac3a619eccb88bd49e9fac41bf2dedf48/braindecode/models/deep4.py#L323

Is there a reason for this? I don't see a flag to load pretrained weights. Is it a convention for other models as well?

**Comment by bruAristimunha** [comment]
By default, when you create, you want to use it with eval to infer and disable the components that don't need to run during this period, e.g. dropout. If you are using skorch, it will change this for you. We are your training model with pure pytorch, you need to use it in the training model and during the validation in eval. 

I think this was the convention that Lukas adopted when he implemented shallownet, tcn and deep4net. I think it used to be more sense before close #393 and the old version of skorch, but nowadays, it doesn't matter that much. 

And what do you mean by load pre-trained weights? It's a pytorch module, so you can use the default load weight.

**Comment by bruAristimunha** [comment]
ok ok, we can change for deep4net, tcn and shallow this behaviour.

**Comment by dungscout96** [comment]
Noted, thanks!

**Comment by bruAristimunha** [comment]
Should be fine now @dungscout96
